includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -581,16 +581,32 @@ public function get_title() {
 		$title = parent::get_title();
 
 		if (
-			Payment_Method::CARD === $this->stripe_id &&
 			( is_checkout() || is_add_payment_method_page() ) &&
 			! isset( $_GET['change_payment_method'] )  // phpcs:ignore WordPress.Security.NonceVerification
 		) {
+			$test_mode_badge = '';
 			if ( WC_Payments::mode()->is_test() ) {
 				$test_mode_badge = '<span class="test-mode badge">' . __( 'Test Mode', 'woocommerce-payments' ) . '</span>';
-			} else {
-				$test_mode_badge = '';
 			}
-			return '<div class="label-title-container"><span class="payment-method-title">&nbsp;' . $title . '</span>' . $test_mode_badge . '</div>';
+
+			$bnpl_messaging_container = '';
+			if ( $this->payment_method->is_bnpl() ) {
+				$bnpl_messaging_container = '<span id="stripe-pmme-container-' . $this->payment_method->get_id() . '" class="stripe-pmme-container"></span>';
+			}
+
+			// the "plain" payment method label is displayed on some sections of the app
+			// - like "pay for order" when a payment method is pre-selected or a payment has previously failed.
+			$html  = '<span class="woopayments-plain-payment-method-label">' . $title . '</span>';
+			$html .= '<div class="woopayments-rich-payment-method-label">';
+			$html .= '<div class="label-title-container">';
+			$html .= '<span class="payment-method-title">&nbsp;' . $title . '</span>';
+			$html .= $test_mode_badge;
+			$html .= '</div>';
+			$html .= $this->get_icon();
+			$html .= $bnpl_messaging_container;
+			$html .= '</div>';
+
+			return $html;
```

Comment:
Changing the previous approach a bit - I'm now rendering two elements: a "rich" and a "plain" element.
It turns out that the pay-for-order page will use this method to render the payment method (if a previous attempt failed).

So I need to render both labels and (via CSS) selectively hide/show the most appropriate one based on the context the label is displayed.